apps/lib/app_provider.c

```diff
@@ -13,6 +13,9 @@
 #include <openssl/provider.h>
 #include <openssl/safestack.h>
 
+/* Non-zero if any of the provider options have been seen */
+static unsigned char provider_option_given = 0;
```

Comment:
I'd probably use just `static int`.

Reply:
This saves a few bytes but we're pretty profligate.  I'll switch.